### PR TITLE
backend mega-refactor

### PR DESCRIPTION
**Pull request by dr-bonez**
- Refactor NetController
- Refactor Account
- Refactor Manager

### CODE REVIEW

File: backend/src/backup/os.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#[derive(Deserialize, Serialize)]
+#[serde(rename = "kebab-case")]
+struct OsBackupV1 {
+    server_id: String,         // uuidv4
+    hostname: String,          // embassy-<adjective>-<noun>
+    net_key: Base64<[u8; 32]>, // Ed25519 Secret Key
+    root_ca_key: String,       // PEM Encoded OpenSSL Key
+    root_ca_cert: String,      // PEM Encoded OpenSSL X509 Certificate
+    ui: Value,                 // JSON Value
+                               // TODO add more
```
Comment (Blu-J) [inline backend/src/backup/os.rs:95]: Still want the todo?

File: backend/src/backup/restore.rs [hunk 2]
```diff
@@ -414,23 +395,32 @@ async fn restore_package<'a>(
...
+        let k = key.0.as_slice();
+        sqlx::query!(
+            "INSERT INTO network_keys (package, interface, key) VALUES ($1, $2, $3) ON CONFLICT (package, interface) DO NOTHING",
+            *id,
+            *iface,
+            k,
+        )
+        .execute(&mut secrets_tx).await?;
+    }
+    // DEPRECATED
```
Comment (Blu-J) [inline backend/src/backup/restore.rs:411]: Should we tie a story to this deprecation?

File: backend/src/manager/transition_state.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+use tokio::task::JoinHandle;
+
+pub(crate) enum TransitionState {
+    // Starting(JoinHandle<()>),
+    // Stopping(JoinHandle<()>)
```
Comment (Blu-J) [inline backend/src/manager/transition_state.rs:5]:
```suggestion
```

**Comment by dr-bonez** [comment]
closing in favor of #2338

**Pull request closed without merging**